getting this error for betacode on python2:
```
builder for '/nix/store/7cc9snhy0kjsskb77mb23fm2z8f03rr9-python2.7-betacode-1.0.drv' failed with exit code 1; last 10 log lines:
  Executing setuptoolsBuildPhase
  Traceback (most recent call last):
    File "nix_run_setup", line 8, in <module>
      exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
    File "setup.py", line 22, in <module>
      long_description = read('README.rst'),
    File "setup.py", line 14, in read
      with open(os.path.join(os.path.dirname(__file__), fn), encoding='utf-8') as f:
  TypeError: 'encoding' is an invalid keyword argument for this function
  builder for '/nix/store/7cc9snhy0kjsskb77mb23fm2z8f03rr9-python2.7-betacode-1.0.drv' failed with exit code 1
cannot build derivation '/nix/store/shllrgfahfx5di0vpkigkq0s5znwncp5-env.drv': 1 dependencies couldn't be built
```
maybe disable it on python2?

--- a/pkgs/development/python-modules/betacode/default.nix
+++ b/pkgs/development/python-modules/betacode/default.nix
@@ -1,0 +1,19 @@
+{ fetchPypi, buildPythonPackage, pygtrie, isPy3k, lib, }:
+buildPythonPackage rec {
+  pname = "betacode";
+  version = "1.0";
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0s84kd9vblbjz61q7zchx64a6hmdqb4lillna5ryh0g9ij76g6r5";
+  };
+  preBuild = ''echo > README.rst'';
+  # setup.py uses a python3 os.path.join
+  disabled = !isPy3k;
+  propagatedBuildInputs = [ pygtrie ];
+  meta = {
+    homepage = "https://github.com/matgrioni/betacode";
+    description = "A small python package to flexibly convert from betacode to unicode and back.";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ kmein ];
+  };
+}

--- a/pkgs/development/python-modules/pygtrie/default.nix
+++ b/pkgs/development/python-modules/pygtrie/default.nix
@@ -1,0 +1,15 @@
+{ lib, fetchPypi, buildPythonPackage, ... }:
+buildPythonPackage rec {
+  pname = "pygtrie";
+  version = "2.3";
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "00x7q4p9r75zdnw3a8vd0d0w0i5l28w408g5bsfl787yv6b1h9i8";
+  };
+  meta = {
+    homepage = "https://github.com/mina86/pygtrie";
+    description = "Trie data structure implementation";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ kmein ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -817,6 +817,8 @@ in {
 
   bespon = callPackage ../development/python-modules/bespon { };
 
+  betacode = callPackage ../development/python-modules/betacode { };
+
   betamax = callPackage ../development/python-modules/betamax { };
 
   betamax-matchers = callPackage ../development/python-modules/betamax-matchers { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -5023,6 +5023,8 @@ in {
 
   pygtksourceview = callPackage ../development/python-modules/pygtksourceview { inherit (pkgs) pkgconfig; };
 
+  pygtrie = callPackage ../development/python-modules/pygtrie { };
+
   pyhamcrest = if isPy3k then
     callPackage ../development/python-modules/pyhamcrest { }
   else

